### PR TITLE
Add Jekyll Example

### PR DESCRIPTION
**Pull request by timothyis**
This pull request adds an example of deploying a Jekyll site that builds in the cloud and serves the static output.

### CODE REVIEW

File: jekyll/readme.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+Alternatively, your new Jekyll site can be automatically deployed and aliased using [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-gitlab). Pushing these files to a new repository with the `now.json` in the root, and with either [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-github) configured for that repository, your site will be automatically deployed for every push and pull/merge request, and aliased for every push to the default branch!
+
+## Included In This Starter
+
+This starter project includes:
+- A fresh Jekyll app using the [Jekyll CLI](https://jekyllrb.com/docs/usage/) (with a few edits to provide resources!)
+- A pre-defined `build.sh` shell script that installs Ruby through `rbenv`, installs the dependencies (including Jekyll), then builds the app to the default `_site` output directory.
+- A pre-configured `now.json` file that uses the aforementioned build shell script and deploy the `_site` directory when built.
```
Comment (msweeneydev) [inline jekyll/readme.md]:
```suggestion
- A pre-configured `now.json` file that uses the aforementioned build shell script and deploys the `_site` directory when built.
```

File: jekyll/readme.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+$ now
+```
+
+Alternatively, your new Jekyll site can be automatically deployed and aliased using [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-gitlab). Pushing these files to a new repository with the `now.json` in the root, and with either [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-github) configured for that repository, your site will be automatically deployed for every push and pull/merge request, and aliased for every push to the default branch!
+
+## Included In This Starter
+
+This starter project includes:
+- A fresh Jekyll app using the [Jekyll CLI](https://jekyllrb.com/docs/usage/) (with a few edits to provide resources!)
+- A pre-defined `build.sh` shell script that installs Ruby through `rbenv`, installs the dependencies (including Jekyll), then builds the app to the default `_site` output directory.
```
Comment (msweeneydev) [inline jekyll/readme.md]:
```suggestion
- A pre-defined `build.sh` shell script that installs Ruby through `rbenv`, the dependencies (including Jekyll), then builds the app to the default `_site` output directory.
```

File: jekyll/readme.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+> Note: Make sure to install the jekyll and bundler gems with `gem install jekyll bundler` in your terminal, then you can install the dependencies before running your development server for the first time with `bundle exec jekyll serve`.
+
+To [deploy](https://zeit.co/docs/v2/deployments/basics) this application, with [Now installed](https://zeit.co/docs/v2/getting-started/installation), run the following from your terminal:
+
+```shell
+$ now
+```
+
+Alternatively, your new Jekyll site can be automatically deployed and aliased using [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-gitlab). Pushing these files to a new repository with the `now.json` in the root, and with either [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-github) configured for that repository, your site will be automatically deployed for every push and pull/merge request, and aliased for every push to the default branch!
```
Comment (msweeneydev) [inline jekyll/readme.md]:
```suggestion
Alternatively, your new Jekyll site can be automatically deployed and aliased using [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-gitlab). Pushing these files to a new repository with `now.json` in the root, and with either [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-github) configured for that repository, your site will be automatically deployed for every push and pull/merge request, and aliased for every push to the default branch!
```

File: jekyll/about.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+[minima](https://github.com/jekyll/minima)
+
+You can find the source code for Jekyll on GitHub:
+[jekyll][jekyll-organization] /
+[jekyll](https://github.com/jekyll/jekyll)
+
+You can find the source code for this Now deployed example on GitHub:
+[now-examples](https://github.com/zeit/now-examples)
+
+You can find more information on ZEIT Now on the ZEIT docs:
```
Comment (msweeneydev) [inline jekyll/about.md]:
```suggestion
You can find more information on ZEIT Now in the ZEIT docs:
```

File: jekyll/_posts/2019-03-26-welcome-to-jekyll.markdown [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+date:   2019-03-26 03:28:01 +0000
+categories: jekyll now deployment example
+---
+
+Welcome to this deployment example of Jekyll built and deployed in the cloud with [ZEIT Now](https://zeit.co/now)!
+
+To find the source code of this deployment and many more deployed application examples, see our [Now Examples repository](https://github.com/zeit/now-examples).
+
+## Usage
+You’ll find this post in the `_posts` directory. Go ahead and edit it and re-build the site to see your changes. You can rebuild the site in many different ways, but the most common way is to run `jekyll serve`, which launches a web server and auto-regenerates your site when a file is updated.
```
Comment (msweeneydev) [inline jekyll/_posts/2019-03-26-welcome-to-jekyll.markdown]:
```suggestion
You’ll find this post in the `_posts` directory. Go ahead and edit it then re-build the site to see your changes. You can rebuild the site in many different ways, the most common way is to run `jekyll serve`, which launches a web server and auto-regenerates your site when a file is updated.
```

File: jekyll/_posts/2019-03-26-welcome-to-jekyll.markdown [hunk 6]
```diff
@@ -0,0 +1,51 @@
+---
+layout: post
+title:  "Welcome to Jekyll on Now!"
+date:   2019-03-26 03:28:01 +0000
+categories: jekyll now deployment example
+---
+
+Welcome to this deployment example of Jekyll built and deployed in the cloud with [ZEIT Now](https://zeit.co/now)!
```
Comment (msweeneydev) [inline jekyll/_posts/2019-03-26-welcome-to-jekyll.markdown]:
```suggestion
Welcome to this deployment example of Jekyll, built and deployed in the cloud with [ZEIT Now](https://zeit.co/now)!
```

File: jekyll/_config.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+# Site settings
+# These are used to personalize your new site. If you look in the HTML files,
+# you will see them accessed via {{ site.title }}, {{ site.email }}, and so on.
+# You can create any custom variable you would like, and they will be accessible
+# in the templates via {{ site.myvariable }}.
+title: Jekyll on ZEIT Now
+email: your-email@example.com
+description: >- # this means to ignore newlines until "baseurl:"
+  This is an example of Jekyll on ZEIT Now. See the sole post on this example for more information, or see the source to learn how to deploy yourself at github.com/zeit/now-examples
```
Comment (msweeneydev) [inline jekyll/_config.yml]:
```suggestion
  This is an example of Jekyll deployed on ZEIT Now. See the sole post on this example for more information, or see the source to learn how to deploy yourself at github.com/zeit/now-examples
```

File: jekyll/readme.md (hunk 3 above)
Comment (timothyis) [reply jekyll/readme.md]: How about:
```suggestion
Alternatively, your new Jekyll site can be automatically deployed and aliased using [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-gitlab). Pushing these files to a new repository with a `now.json` file in the root, and with either [Now for GitHub](https://zeit.co/docs/v2/integrations/now-for-github) or [Now for GitLab](https://zeit.co/docs/v2/integrations/now-for-github) configured for that repository, your site will be automatically deployed for every push and pull/merge request, and aliased for every push to the default branch!
```

**Comment by travishines** [comment]
@timothyis — My `build.sh` seems to fail at the Building stage of using `now`. My log shows bundler successfully installing, then — https://gist.github.com/travishines/64f4cc33304373708b7c42b685f766cd

**Comment by travishines** [comment]
@timothyis — It's been hit or miss for my `now` deployment at the building stage of `build.sh`. The couple times it has worked for me has taken several minutes. The failed builds all seem to share the same error after successfully installing bundler — https://gist.github.com/travishines/64f4cc33304373708b7c42b685f766cd

Side note: the few successful builds I've had lack support for Jekyll permalinks. Ex. A post with the permalink set to `/example` returns a 404, but `/example.html` works. Not sure the cause of this new issue.

**Comment by igorklopov** [comment]
@travishines could you please post full log around https://gist.github.com/travishines/64f4cc33304373708b7c42b685f766cd and/or give deployment url.

**Comment by travishines** [comment]
@igorklopov — here's my full log from my last failed deploy (2 hours ago): https://gist.github.com/travishines/6b68b27c597ba259097e680f4a1da897

**Comment by igorklopov** [comment]
@travishines thanks you! looks like it is another error (different from https://gist.github.com/travishines/64f4cc33304373708b7c42b685f766cd). anyways looking at both

**Comment by igorklopov** [comment]
@travishines Seems like it is a known issue https://bundler.io/blog/2019/01/04/an-update-on-the-bundler-2-release.html. I have pinned `bundler` version and removed lock file (https://github.com/zeit/now-examples/commit/d0d2635524bff16714cf1e0645c7bc45b6e4091b). Can you please try again with the updated example. Meanwhile i will try to craft a better solution, if there is any.

**Comment by travishines** [comment]
@igorklopov — using your example (d0d2635) doesn't seem to speed up the process.

Being that `jekyll serve` builds my site locally, would there be a simpler version of `now.json` that just deploys the `_site' folder without worrying about setting up rails and rebuilding the Jekyll site?

**Comment by Ehbraheem** [comment]
@travishines you can use this `now.json` (If you're using @timothyis 's example)  while @igorklopov take a deeper look into the issue.
```json
{
  "version": 2,
  "name": "jekyll",
  "builds": [
    { "src": "build.sh", "use": "jekyll-builder", "config": { "distDir": "_site" } }
  ]
}
```
*Another strict requirement*. Your `build.sh` should be empty unless you are doing something different from installing `ruby` and setting up `bundler` and `jekyll`

This was deployed with the above `now.json` example [https://jekyll.ehbraheem.now.sh/](https://jekyll.ehbraheem.now.sh/)

**Comment by travishines** [comment]
@Ehbraheem — Your instructions seem to work just like @igorklopov's. I think the remaining oddity is that it consistently takes 6 minutes to deploy the site via `now`. Here are my steps for reference:

- I cloned @timothyis's /jekyll folder locally, this time into a new folder to ensure my personal site structure wasn't causing the issue. 
- I deleted everything in the 'build.sh' file, leaving it empty
- I copied your `now.json` code [from above](https://github.com/zeit/now-examples/pull/329#issuecomment-477763578) which has `jekyll-builder` for the `use`. 
- I deployed the site using `now`
- The site deployed to https://jekyll-4tn23bl8b.now.sh on Zeit...
- ...but then it gets hung up, consistently for 6 minutes, `Building` from `build.sh` (which is just empty now)
- The site does eventually deploy (after 6 mins) and can be found at https://jekyll.travishines.now.sh/

**Comment by timothyis** [comment]
Unfortunately, this is a side effect of having to install Ruby and its dependencies @travishines. I am actively speaking to @igorklopov, however, to see how we can speed this up, like for example caching ruby so it doesn't need to install on each deployment.

**Comment by timothyis** [comment]
Update @travishines, I've just updated the build script with a faster method, albeit with an older version of Ruby, which should be fine. This cuts build times from 6 minutes down to 2 minutes.

Would you mind trying it out? https://github.com/zeit/now-examples/commit/df7e6212b90ca770b686ac3071327d6f545b82b3

**Comment by travishines** [comment]
I was still seeing 6 min builds with your df7e621. Then I ran `bundle update` and tried `now` again. Now it deploys in 2 seconds (!!!).

`build.sh` no longer seems to be a step in the process, with no relative delay indicated:

```
zeit now
> Deploying ~/Code/zeit under travishines
> Using project jekyll
> https://jekyll-qd60ly8q3.now.sh [v2] [2s]
> Ready! Aliased to https://jekyll.travishines.now.sh [in clipboard] [2s]
```

Thanks so much for working through this with me, @timothyis @Ehbraheem @igorklopov.

If I'm missing any details, or if there's anything else I can do to help with your support for Jekyll, let me know! :)

**Comment by timothyis** [comment]
@travishines you're seeing 2s deploy times in subsequent attempts due to [deduplication](https://zeit.co/docs/v2/deployments/concepts/immutability/#deduplication-algorithm). You can force a new deployment, if needed, using `now -f`.

I'm surprised to see that the new build script took 6 minutes still, though. Could you try a new deployment using the build script and `@now/static-build` in your `now.json` instead of `jekyll-builder`? It should speed it up quite a lot.

Additionally, we are in progress with some improvements to deployment dependencies that should make this process even faster.

**Comment by travishines** [comment]
@timothyis — using `@now/static-build` and `now -f` forcing a build clocks in at 2 minutes consistently 👍.

**Comment by timothyis** [comment]
Great, @travishines! I'm glad we cut it down that much already. We still working to trim the time it takes to deploy even more, however. Excited to share updates with you.

If you would like to take part in testing or contributing to progress such as this and beta features, we have a public #education Slack channel that you are invited to. Let me know if you'd like to join. (Just send me your email over twitter DMs and I'll invite you).

**Pull request merged**